### PR TITLE
Fixed forms not updated when dashboard updates form

### PR DESCRIPTION
**Pull request by georific**
## Status
**READY**

## JIRA Ticket
SPACON-661

## Description
Fixed forms not updating when dashboard updates form

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] License

## Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

```sh
git fetch --all
git checkout <feature_branch> 
./gradlew connectedCheck
```

@boundlessgeo/spatial-connect

### CODE REVIEW

File: spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/db/GeoPackage.java [hunk 1]
```diff
@@ -704,9 +763,49 @@ public void addFeatureSource(String layer, Map<String,String>  fields) {
...
+                for (Map.Entry<String, String> col : newSchemaCols.entrySet())
+                {
+                    cursor = db.query(String.format("ALTER TABLE %s ADD COLUMN %s %s",
+                            auditTableName, col.getKey(), col.getValue())
+                    );
+                    cursor.moveToFirst();
+                }
+                tx.markSuccessful();
+            } catch (Exception e) {
+                Log.e(LOG_TAG, "Error adding to existing audit table: " + e.getMessage());
```
Comment (tetriscode) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/db/GeoPackage.java]: Do you need to abort the transaction in this catch?

File: spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/db/GeoPackage.java [hunk 2]
```diff
@@ -704,9 +763,49 @@ public void addFeatureSource(String layer, Map<String,String>  fields) {
...
+                    cursor.close();
+                }
+            }
+
         }
     }
 
+    public void addFeatureSource(String layer, Map<String,String>  fields) {
+        addContentsTable(layer, fields);
+        addAuditTable(layer, fields);
```
Comment (tetriscode) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/db/GeoPackage.java]: `addAuditTable` should probably only run if `addContentsTable` is successful. Looks like Audit could run if the contents function fails.

**Review by tetriscode — CHANGES_REQUESTED** [review]

File: spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/db/GeoPackage.java (hunk 1 above)
Comment (georific) [reply spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/db/GeoPackage.java]: the final block has `tx.end()` which will either commit or rollback based on if the tx was marked successfully or not.

File: spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/db/GeoPackage.java [hunk 3]
```diff
@@ -704,9 +763,49 @@ public void addFeatureSource(String layer, Map<String,String>  fields) {
...
+                }
+
+                for (Map.Entry<String, String> col : newSchemaCols.entrySet())
+                {
+                    cursor = db.query(String.format("ALTER TABLE %s ADD COLUMN %s %s",
+                            auditTableName, col.getKey(), col.getValue())
+                    );
+                    cursor.moveToFirst();
+                }
+                tx.markSuccessful();
```
Comment (frankrowe) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/db/GeoPackage.java]: if the audit table already exists, the audit trigger needs to be dropped and recreated.

**Review by frankrowe — CHANGES_REQUESTED** [review]

**Pull request merged**